modules/util/source-formatter/src/test/java/com/liferay/source/formatter/processor/UpgradeCatchAllCheckTest.java

```diff
@@ -0,0 +1,129 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.processor;
+
+import com.liferay.portal.kernel.util.ListUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.SourceFormatterArgs;
+import com.liferay.source.formatter.check.UpgradeCatchAllCheck;
+
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * @author Kyle Miho
+ */
+@RunWith(Parameterized.class)
+public class UpgradeCatchAllCheckTest extends BaseSourceProcessorTestCase {
+
+	@Parameterized.Parameters(name = "Testcase-{index}: Testing {0}")
+	public static Iterable<Object[]> data() throws Exception {
+		String issueKey = System.getProperty("issue.key", null);
+
+		if ((issueKey != null) && !issueKey.isEmpty()) {
+			return ListUtil.fromArray(
+				new Object[][] {{System.getProperty("issue.key")}});
+		}
+
+		List<Object[]> tickets = new ArrayList<>();
+
+		try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(
+				Paths.get(_TESTCASE_WORKING_DIRECTORY))) {
+
+			for (Path path : directoryStream) {
+				String fileName = path.getFileName(
+				).toString();
+
+				fileName = StringUtil.replace(fileName, "LPD", "LPD-");
+
+				tickets.add(
+					new Object[] {
+						StringUtil.removeSubstring(fileName, ".testjava")
+					});
```

Comment:
If we are referencing the json file, we also have access to the type here.
```
".test" + filetype
```

Reply:
We use file type as a param to check for unit tests now as well